src/pool-bin/BinPoolManager.sol

```diff
@@ -264,10 +259,7 @@ contract BinPoolManager is IBinPoolManager, ProtocolFees, Extsload {
         BalanceDelta hookDelta;
         (delta, hookDelta) = BinHooks.afterBurn(key, params, delta, hookData);
 
-        if (hookDelta != BalanceDeltaLibrary.ZERO_DELTA) {
-            vault.accountAppBalanceDelta(key.currency0, key.currency1, hookDelta, address(key.hooks));
-        }
-        vault.accountAppBalanceDelta(key.currency0, key.currency1, delta, msg.sender);
+        vault.accountAppDeltaWithHookDelta(key, delta, hookDelta, msg.sender);
```

Comment:
Do we really need to pass in msg.sender here ? Why not use it from library directly

Reply:
no strong preference, my thinking was around future 'app' which might not want to accountAppDelta with `msg.sender` 

will remove the passing of `msg.sender`

Reply:
updated and it saves gas ❤️